interfaces/seccomp/backend_test.go

```diff
@@ -0,0 +1,257 @@
+	s.repo = interfaces.NewRepository()
+	s.iface = &interfaces.TestInterface{InterfaceName: "iface"}
+	err = s.repo.AddInterface(s.iface)
+	c.Assert(err, IsNil)
+}
+
+func (s *backendSuite) TearDownTest(c *C) {
+	dirs.SetRootDir("/")
+}
+
+// Tests for Configure() and Deconfigure()
+const sambaYamlV1 = `
+name: samba
+version: 1
+developer: acme
+apps:
+    smbd:
+slots:
+    iface:
+`
+const sambaYamlV1WithNmbd = `
+name: samba
+version: 1
+developer: acme
+apps:
+    smbd:
+    nmbd:
+slots:
+    iface:
+`
+const sambaYamlV2 = `
+name: samba
+version: 2
+developer: acme
+apps:
+    smbd:
+slots:
+    iface:
+`
+
+func (s *backendSuite) TestInstallingSnapWritesProfiles(c *C) {
+	developerMode := false
+	s.installSnap(c, developerMode, sambaYamlV1)
+	profile := filepath.Join(dirs.SnapSeccompDir, "snap.samba.smbd")
+	// file called "snap.sambda.smbd" was created
+	_, err := os.Stat(profile)
+	c.Check(err, IsNil)
+}
+
+func (s *backendSuite) TestRemovingSnapRemovesProfiles(c *C) {
+	for _, developerMode := range []bool{true, false} {
+		snapInfo := s.installSnap(c, developerMode, sambaYamlV1)
+		s.removeSnap(c, snapInfo)
+		profile := filepath.Join(dirs.SnapSeccompDir, "snap.samba.smbd")
+		// file called "snap.sambda.smbd" was removed
+		_, err := os.Stat(profile)
+		c.Check(os.IsNotExist(err), Equals, true)
+	}
+}
+
+func (s *backendSuite) TestUpdatingSnapToOneWithMoreApps(c *C) {
+	for _, developerMode := range []bool{true, false} {
+		snapInfo := s.installSnap(c, developerMode, sambaYamlV1)
+		snapInfo = s.updateSnap(c, snapInfo, developerMode, sambaYamlV1WithNmbd)
+		profile := filepath.Join(dirs.SnapSeccompDir, "snap.samba.nmbd")
+		// file called "snap.sambda.nmbd" was created
+		_, err := os.Stat(profile)
+		c.Check(err, IsNil)
+		s.removeSnap(c, snapInfo)
+	}
+}
+
+func (s *backendSuite) TestUpdatingSnapToOneWithFewerApps(c *C) {
+	for _, developerMode := range []bool{true, false} {
+		snapInfo := s.installSnap(c, developerMode, sambaYamlV1WithNmbd)
+		snapInfo = s.updateSnap(c, snapInfo, developerMode, sambaYamlV1)
+		profile := filepath.Join(dirs.SnapSeccompDir, "snap.samba.nmbd")
+		// file called "snap.sambda.nmbd" was removed
+		_, err := os.Stat(profile)
+		c.Check(os.IsNotExist(err), Equals, true)
+		s.removeSnap(c, snapInfo)
+	}
+}
+
+func (s *backendSuite) TestRealDefaultTemplateIsNormallyUsed(c *C) {
+	snapInfo, err := snap.InfoFromSnapYaml([]byte(sambaYamlV1))
+	c.Assert(err, IsNil)
+	// NOTE: we don't call seccomp.MockTemplate()
+	err = s.backend.Configure(snapInfo, false, s.repo)
+	c.Assert(err, IsNil)
+	profile := filepath.Join(dirs.SnapSeccompDir, "snap.samba.smbd")
+	data, err := ioutil.ReadFile(profile)
+	c.Assert(err, IsNil)
+	for _, line := range []string{
+		// NOTE: a few randomly picked lines from the real profile.  Comments
+		// and empty lines are avoided as those can be discarded in the future.
+		"deny kexec_load\n",
+		"deny create_module\n",
+		"symlink\n",
+		"socket\n",
```

Comment:
Thanks for explaining and correcting this.

I've applied your suggestion in 7477864ea90fe9fcf4387ba43f4c21befe556daf

Given the TODO list in ``combineSnippets`` I was wondering if it makes more sense to put more snippet post-processing in the launcher or in snappy. Snappy is arguably easier to get this right (go is a higher-level language than C) but this might collide with some other plans you have for the launcher. Please advise, I'll either remove the TODO list or I'll work on implementing it.